src/Umbraco.Web/Mvc/RenderActionInvoker.cs

```diff
@@ -31,16 +32,15 @@ protected override ActionDescriptor FindAction(ControllerContext controllerConte
                 //check if the controller is an instance of IRenderMvcController
 				if (controllerContext.Controller is IRenderMvcController)
 				{
-				    return IndexDescriptors.GetOrAdd(controllerContext.Controller.GetType(),
-				        type => new ReflectedActionDescriptor(
-				            controllerContext.Controller.GetType().GetMethods()
-				                .First(x => x.Name == "Index" &&
-				                            x.GetCustomAttributes(typeof (NonActionAttribute), false).Any() == false),
-				            "Index",
-				            controllerDescriptor));
-
+					var methodInfo = controllerContext.Controller.GetType().GetMethods()

```

Comment:
The problem with putting all of this logic here is that it totally bypasses the caching in the concurrent dictionary, this logic needs to be wrapped int eh concurrent dictionary callback.

I will fix this.